opm/output/data/Wells.hpp

```diff
@@ -120,13 +122,26 @@ namespace Opm {
         void read(MessageBufferType& buffer);
     };
 
+    struct Segment {
+        Rates rates;
+        double pressure;
+        std::size_t segNumber;
+
+        template <class MessageBufferType>
```

Comment:
Is it *really* necessary to make a template the `MessageBufferType` - without having looked through the rest of the code this seems a bit like premature abstraction to me?

Reply:
I don't really know.  This is mostly to integrate into the existing `read()/write()` mechanism of `data::WellRates`.  I don't know why that exists.